tensorflow/python/ops/math_ops.py

```diff
@@ -1577,6 +1577,25 @@ def sigmoid(x, name=None):
     return gen_math_ops._sigmoid(x, name=name)
 
 
+def sigmoid_grad(x, name=None):
+  """Computes the gradient of the sigmoid of `x` element-wise.
+
+  Specifically, `y = sigmoid(x) * (1 - sigmoid(x))`.
+
+  Args:
+    x: A Tensor with type `float32`, `float64`, `int32`, `complex64`, `int64`,
+      or `qint32`.
+    name: A name for the operation (optional).
+
+  Returns:
+    A Tensor with the same type as `x` if `x.dtype != qint32`
+      otherwise the return type is `quint8`.
+  """
+  with ops.name_scope(name, "SigmoidGrad", [x]) as name:
+    x = ops.convert_to_tensor(x, name="x")
+    return gen_math_ops._sigmoid_grad(x, name=name)
+
```

Comment:
@girving, ok!  I changed the t * t and I will look at writing the tests
tonight.  thanks!

On Mon, Aug 15, 2016 at 8:47 AM, Geoffrey Irving <notifications@github.com>
wrote:

> In tensorflow/python/ops/math_ops.py
> <https://github.com/tensorflow/tensorflow/pull/3807#discussion_r74782183>:
>
> > +
> > +  Specifically, `y = sigmoid(x) * (1 - sigmoid(x))`.
> > +
> > +  Args:
> > +    x: A Tensor with type `float32`, `float64`, `int32`, `complex64`, `int64`,
> > +      or `qint32`.
> > +    name: A name for the operation (optional).
> > +
> > +  Returns:
> > +    A Tensor with the same type as `x` if `x.dtype != qint32`
> > +      otherwise the return type is `quint8`.
> > +  """
> > +  with ops.name_scope(name, "SigmoidGrad", [x]) as name:
> > +    x = ops.convert_to_tensor(x, name="x")
> > +    return gen_math_ops._sigmoid_grad(x, name=name)
> > +
>
> Note that sigmoid_grad and tanh_grad will need separate tests too (to
> verify that they are indeed the gradients named.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/3807/files/e83b0b03eb93c60b1aec6733c147eb24e25fa0ed#r74782183>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADzDDL4jkmySCAT3Y8d1B0Fag88w5bpYks5qgIoRgaJpZM4Jj8mC>
> .
>